arviz/data/inference_data.py

```diff
@@ -412,6 +415,10 @@ def to_netcdf(
             Location of netcdf file
         """
         mode = "w"  # overwrite first, then append
+        if self._attrs:
+            xr.Dataset(attrs=self._attrs).to_netcdf(filename, mode=mode, group="__netcdf4_attrs")
```

Comment:
```suggestion
            xr.Dataset(attrs=self._attrs).to_netcdf(filename, mode=mode)
```

xarray is able to both read and write datasets that only have attributes, and if no group is provided then things are saved at the root level which is what we want here.

I have double checked and I can also use netcdf4 library directly to get the saved attributes correctly with `ncattrs`, so I think it should be possible to do the same in the other languages. 

Moreover, xarray is also able to read//write datasets that only have attrs to the root level using `to_zarr` and `open_zarr`, so I think using this will also simplify the code

Reply:
If this works, great news.

I tried and just failed with netcdf4 lib, but I didn't dataset. I will neeed to try again.